Lib/Gateway/Paypal.php

```diff
@@ -1292,8 +1289,19 @@ public function prepare_to_send( $data ) {
 
             $data['subtotal'] = $billing_amount - $tax_amount;
             $data['tax'] = $tax_amount;
+
+            // Store the correctly calculated billing amount before filters
+            $correct_billing_amount = $billing_amount;
+
+            // Apply filters but ensure we use the correct amount if tax was already calculated
             $billing_amount = apply_filters( 'wpuf_payment_amount', $billing_amount, $post_id );
```

Comment:
we need to check from where the `$billing_amount` is getting modified using `wpuf_payment_amount` filter. if we revert the amount, the filter hook looses it's purpose